src/MultiSelect/MultiSelect.tsx

```diff
@@ -0,0 +1,235 @@
+
+  const {
+    isOpen,
+    getToggleButtonProps,
+    getLabelProps,
+    getMenuProps,
+    getInputProps,
+    getComboboxProps,
+    getItemProps,
+    highlightedIndex,
+    openMenu,
+  } = useCombobox({
+    inputValue,
+    defaultHighlightedIndex: 0,
+    // useMultipleSelection will separately take care of all "selected item" interactions
+    // without this line, interaction with adding/removing will be buggy
+    selectedItem: null,
+    items: selectableItems,
+    // taken from https://www.downshift-js.com/use-multiple-selection
+    stateReducer: (state, actionAndChanges) => {
+      const { changes, type } = actionAndChanges;
+      switch (type) {
+        case useCombobox.stateChangeTypes.InputKeyDownEnter:
+        case useCombobox.stateChangeTypes.ItemClick:
+          return {
+            ...changes,
+            isOpen: true, // keep the menu open after selection.
+          };
+        default:
+          break;
+      }
+      return changes;
+    },
+    // default built-in changes generated here
+    // https://github.com/downshift-js/downshift/blob/a13308910282c44ea2ebdcbc8a6dfee901df48a2/src/hooks/useCombobox/reducer.js
+    // taken from https://www.downshift-js.com/use-multiple-selection
+    onStateChange: (changes) => {
+      const { inputValue: value, type, selectedItem } = changes;
+      switch (type) {
+        case useCombobox.stateChangeTypes.InputChange:
+          setInputValue(value);
+          break;
+        case useCombobox.stateChangeTypes.InputKeyDownEnter:
+        case useCombobox.stateChangeTypes.ItemClick:
+          if (selectedItem) {
+            setInputValue("");
+            addSelectedItem(selectedItem);
+          }
+          break;
+        default:
+          break;
+      }
+    },
+  });
+
+  const id = name;
+  return (
+    <div className={classNames(cssClass.CONTAINER, formElementSizeClassName(size), className)}>
+      <label
+        className={classNames(cssClass.LABEL, hideLabel && cssClass.LABEL_HIDDEN)}
+        {...getLabelProps()}
+      >
+        {label}
+      </label>
+      <div
+        className={classNames(
+          cssClass.SELECT_CONTAINER,
+          isOpen && cssClass.SELECT_CONTAINER_FOCUSED,
+        )}
+        {...getComboboxProps()}
+      >
+        <div className={cssClass.SELECTED_ITEMS_CONTAINER}>
+          {selectedItems.length > 0 &&
+            selectedItems.map((item, i) => (
+              <Label
+                key={`${item.key}${i}`}
+                className={cssClass.SELECTED_ITEM_CONTAINER}
+                {...getSelectedItemProps({ selectedItem: item, index: i })}
+              >
+                {item.value || item.stringValue}
+                <span
+                  className={cssClass.SELECTED_ITEM_BUTTON}
+                  onClick={(e) => {
+                    e.stopPropagation();
+                    removeSelectedItem(item);
+                  }}
+                >
+                  &#10005;
```

Comment:
Is there an advantage to using the code rather than the symbol here? If we need to use the symbol, my preference would be to include the name as a comment or declare it as a constant.

Reply:
Whoops, this is straight from the downshift example. I'm guessing they did it for readability, but I can add a comment as well